miniconf_derive/src/lib.rs

```diff
@@ -58,7 +58,7 @@ fn get_path_arm(struct_field: &StructField) -> proc_macro2::TokenStream {
                 if peek {
                     Err(miniconf::Error::PathTooLong)
                 } else {
-                    Ok(miniconf::serde_json_core::to_slice(&self.#match_name, value)?)
+                    serde::ser::Serialize::serialize(&self.#match_name, ser).map_err(|_| miniconf::Error::Serialization)
```

Comment:
We need to fully qualify the `serde` path to something like `miniconf::serde` - `serde` may not be a direct dependent in the user crate.